packages/remix-dev/compiler.ts

```diff
@@ -311,7 +311,7 @@ async function buildEverything(
 
 async function createBrowserBuild(
   config: RemixConfig,
-  options: BuildOptions & { incremental?: boolean }
+  options: Required<BuildOptions> & { incremental?: boolean }
```

Comment:
Is there a reason why you've put `Required` around this?

This would technically be a breaking change.

Reply:
Well, taking a look at the code we can see that the method
`createBrowserBuild` is invoked by `buildEverything` (reference: image below), for this method the `option` parameter is _required_.

So, we have two methods: **A** that invoke **B**.
1. Method A treats parameter X as required;
2. Method A sends the same parameter received directly to method B;
2.1. So method B should receive something in the same format, as "required" (right?);
3. Therefore, if method A does not receive something that is `required` it should throw an error and will never get to invoke method B.


<img width="566" alt="image" src="https://user-images.githubusercontent.com/5417662/160297442-7084ef5e-2cf6-40d1-9838-d9db81a8fcdf.png">


@MichaelDeBoey Does this make sense?

Reply:
It doesn't matter that the parent function (`buildEverything`) is more strict than the child function (`createBrowserBuild`). If it would be the other way around, that would indeed be a problem.

Since `createBrowserBuild` is only used in `buildEverything`, it's fine to make `createBrowserBuild` more strict too. 👍